libr/bin/blang.c

```diff
@@ -106,17 +109,17 @@ R_API int r_bin_load_languages(RBinFile *binfile) {
 		}
 		if (!cantbe.cxx) {
 			bool hascxx = false;
-			if (unknownType || !(isMacho || isElf)) {
-				cantbe.swift = false;
-				continue;
```

Comment:
I think the `if (unknownType || ...)` statements should move out of the loop because the tested variables are not changed during the loop.

Reply:
I just found out that the `r_list_foreach (o->libs, iter2, lib)` loops are inside the foreach(symbols) loop, which I think also very strange.

Reply:
Well, the. idea is that swift, rust and c++ have similar constructions in the symbols, and detecting sometimes requires disabling c++ check when finding out it's swift. that's why it was not outside the loop. This can be tricky so the proper way would be to make 2 passes, but this will be too slow and we can work on the false positives if any

Reply:
i dont think we have tests for this, but we should

Reply:
OK, so I'll just keep the o->libs loop. I think ``(unknownType || ...)`` can be moved out of the loop, and skip all the checks if this condition is true.